internal/dlw/delve.go

```diff
@@ -0,0 +1,59 @@
+package dlw
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-delve/delve/cmd/dlv/cmds"
+
+	"github.com/juju/juju/internal/dlw/config"
+)
+
+// MainWithArgs is a type representing a function that takes command-line arguments as input and returns an exit code.
+type MainWithArgs func(args []string) int
+
+const (
+	// NoDebug is an environment variable used to redirect the control flow to a new delve instance
+	// or to the program. It is set as environment variable the first time the flow pass in the function
+	NoDebug = "DELVE_ANYTHING_NO_DEBUG"
```

Comment:
After double checking the code, this const doesn't meant to be exported. The purpose of the package is to hide its implementation, so there is no reason to expose the constant.
 
I fact, I see only one reason to expose the content of this const (not the name), it would be to allows someone which is not happy with the debug configuration  to avoid running juju in debug mode, even if he had build it in debug mode. I don't think it would even be a use case.

So the name doesn't matter as long as it is not exported, as long as the content as long as it doesn't collapse with some existing environment variable (as far as I know `os.SetEnv` will set the environmement variable in the shell context, and not export any variable).

Regarding this though, i just unexported it. If there is use cases where we want actually set this env outside the lib, the exact name should be a part of the configuration.